tests/test_tag.py

```diff
@@ -155,3 +155,44 @@ def test_push_tag_failed(mock_repo):
     with patch('git.repo.fun.name_to_object'):
         with pytest.raises(exceptions.PushException):
             repo.tag.push("my_tag", "origin")
+
+
+def test_list_tags(mock_repo):
+    """
+    GIVEN GitRepo is initialized with a path and repo
+    WHEN GitTag.list_tags() is called
+    THEN repo.tags is returned
+    """
+    class test_tag(object):
+
+        def __init__(self, name):
+            self.name = name
+
+    tag1 = test_tag('tag1')
+    tag2 = test_tag('tag2')
+    expected = [tag1, tag2]
+    expected_names = [x.name for x in expected]
+
+    type(mock_repo).tags = PropertyMock(return_value=expected)
+
+    repo = GitRepo(repo=mock_repo)
+    test_tags = repo.tag.list_tags()
+
+    assert expected_names == test_tags
+
+
+def test_list_tags_failed(mock_repo):
+    """
+   GIVEN GitRepo is initialized with a path and repo
+   WHEN GitTag.list_tags() is called
+   AND repo.tags fails
+   THEN a TaggingException is raised
+   """
+    def func():
+        raise git.GitCommandError('tags', '')
+
+    type(mock_repo).tags = PropertyMock(side_effect=func)
```

Comment:
I think the extra function is unnecessary, `side_effect=git.GitCommandError('tags', '')` gives similar results in my testing.